examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -65,23 +81,33 @@ NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
     goto out;
   }
 
+  struct bcc_symbol resolved_symbol;
+  if (NativeStackTrace::insert_dso_name) {
+    procSymbols = get_proc_symbols(pid);
+  }
+
   do {
     unw_word_t offset;
-    char sym[256];
-    char   *realname;
-    int     status;
-
     // TODO: This function is very heavy. We should try to do some caching here, maybe in the
     //       underlying UPT function.
     res = unw_get_proc_name(&cursor, sym, sizeof(sym), &offset);
     if (res == 0) {
-      realname = abi::__cxa_demangle(sym, NULL, NULL, &status);
-      if (status == 0) {
-        this->symbols.push_back(std::string(realname));
+      unw_word_t ip;
+      unw_get_reg(&cursor, UNW_REG_IP, &ip);
+      if (procSymbols && procSymbols->resolve_addr(ip, &resolved_symbol, true)) {
```

Comment:
Let me make sure I understand the logic (and in any case, please document it here). What we do now is:
1. Unwind via libunwind
2. Per IP, if DSO names are enabled, we attempt to symbolicate with `ProcSyms`. If it succeeds we add a symbol with the demangled name + DSO name.
3. Per IP, if DSO names are disabled OR `ProcSyms` symbolication failed, we fall back to the old method of libunwind + `__cxa_demangle`.
Correct?

Reply:
Yes, that's right.

Reply:
After latest commit including suggestion https://github.com/Granulate/bcc/pull/38#discussion_r1020957591 the logic is as follows:
1. If DSO names are enabled, attempt to symbolicate with `ProcSyms`. If it succeeds we add a symbol with the demangled name + DSO name.
2. If step 2. failed, or DSO was not enabled, Unwind via `libunwind` + `__cxa_demangle`.

Reply:
Kk good